### ISSUE TITLE
Provide fresh builds

### ISSUE THREAD

**Issue by ngirard** [ad-si/TaskLite#47]
Hey there,

I just stumbled upon your project and wish to give it a try, but since the latest GH action builds are old, its artefacts are gone.
Would you mind trigger a fresh build ?

Also, I was wondering what's the roadmap for the near future, if any ? The FAQ mentions a [Roadmap](https://tasklite.org/roadmap) page but browsing it returns a 404.

Thanks in advance, and thank you anyway for your time and dedication into this nice project !

**Comment by ad-si** [comment]
Hi, thanks for the heads up. I didn't know that artifacts could expire 😳. But makes sense I guess … otherwise GitHub would have to store *a lot* of data 😅. Looks like'll have to use GitHub releases after all 🙄.
Just pushed some changes to trigger the new build, but it seems to be broken right now. I'll have a look at it the coming days. In the meantime you could use the Docker image: https://tasklite.org/installation/cli.html#prebuilt-docker-image. Works actually really well.

> Also, I was wondering what's the roadmap for the near future, if any ? The FAQ mentions a Roadmap page but browsing it returns a 404.

Good point. I will write a few words. 
TLDR: I'm still actively developing it, although with a reduced velocity due to my workload at @feramhq.

> Thanks in advance, and thank you anyway for your time and dedication into this nice project !

Thanks for the kind words 😊

**Comment by ad-si** [comment]
Added a [roadmap page](https://tasklite.org/roadmap.html) and answered all FAQs 🚀

**Comment by ngirard** [comment]
Thanks for the readmap page, nice read !

Your documentation is very well written, by the way, and the detailed list of similar tools was much appreciated.
I'm reassured to hear that you're still active on your project. 

To be honest, I'm a bit afraid of adopting Tasklite because my literacy in Haskell wouldn't allow me to hack it. On the other hand, the architecture you chose, i.e. client/server with SQLite as a backend, makes so much more sense to me, and would give me a full control over my data.

How feasible would it be to write a Taskwarrior data importer/exporter, in your opinion ? Have you ever though about doing so ?
If such tool existed, it would allow to use Tasklike while benefiting from the wide ecosystem surrounding Taskwarrior when needed. I'm thinking about Timewarrior for time tracking, and another tool written in Python, whose name I forgot, that allows to collect actionable tasks from many web services.

**Comment by ad-si** [comment]
> Thanks for the roadmap page, nice read !
> Your documentation is very well written, by the way, and the detailed list of similar tools was much appreciated.

Thank you ☺️

> because my literacy in Haskell wouldn't allow me to hack it

I understand that concern, but you seem to be into Rust, right? It should then be easy to get into Haskell, if you want to. Others have already written about that transition: http://xion.io/post/programming/rust-into-haskell.html
I'd definitely recommend it. Learning Haskell has been one of the most fulfilling endeavours in my professional career. 

> How feasible would it be to write a Taskwarrior data importer/exporter

Importing tasks from Taskwarrior is already supported: https://tasklite.org/import.html#taskwarrior
The other direction should also be quite easy with TaskLite's JSON export and Taskwarrior's JSON import. Might need a little rewriting with [jq](https://stedolan.github.io/jq/), but apart from that it should be pretty straight forward.

**Comment by ad-si** [comment]
Fixed the build and artifacts are available again: https://github.com/ad-si/TaskLite/actions/runs/712783692 😁

**Comment by ngirard** [comment]
> Fixed the build and artifacts are available again

Thanks for your efforts !
Meanwhile, I've been working a bit on the Docker image. I'm up to something, I just want to be sure that some message I'm getting during the build is not problematic. I'll submit a PR when I'm 100% comfortable with it.

> Importing tasks from Taskwarrior is already supported: https://tasklite.org/import.html#taskwarrior

Wow nice, I hadn't read that part of the docs !

> The other direction should also be quite easy with TaskLite's JSON export and Taskwarrior's JSON import. Might need a little rewriting with jq, but apart from that it should be pretty straight forward.

This would be useful. I'm about to file a dedicated issue.

> you seem to be into Rust, right? It should then be easy to get into Haskell, if you want to. Others have already written about that transition: http://xion.io/post/programming/rust-into-haskell.html

Yes, I read that article back in 2019, I should probably re-read it.
I actually started to learn Haskell, but didn't work on any real project so it didn't stick and I've probably forgotten everything.
But yeah, Haskell is a great language, and your enthusiasm about it is catching... Hacking TaskLite might be an excellent opportunity for me to get back to it.
Would you be willing to mentor me for a few hours here and there in the near future ?

> I'd definitely recommend it. Learning Haskell has been one of the most fulfilling endeavours in my professional career.

This is much interesting. Could you please elaborate ? Also, which benefits do you personally take from Haskell, compared to Rust ?
